qiskit_addon_dice_solver/dice_solver.py

```diff
@@ -46,6 +47,121 @@ def __init__(self, command, returncode, log_path):
+            values such that each row represents a single bitstring. The spin-up configurations
+            should be specified by column indices in range ``(N, N/2]``, and the spin-down
+            configurations should be specified by column indices in range ``(N/2, 0]``, where ``N``
+            is the number of qubits.
+        hcore: Core Hamiltonian matrix representing single-electron integrals
+        eri: Electronic repulsion integrals representing two-electron integrals
+        mpirun_options: Options controlling the CPU resource allocation for the ``Dice`` command line application.
+            These command-line options will be passed directly to the ``mpirun`` command line application during
+            invocation of ``Dice``. These may be formatted as a ``Sequence`` of strings or a single string. If a ``Sequence``,
+            the elements will be combined into a single, space-delimited string and passed to
+            ``mpirun``. If the input is a single string, it will be passed to ``mpirun`` as-is. If no
+            ``mpirun_options`` are provided by the user, ``Dice`` will run on a single MPI slot. For more
+            information on the ``mpirun`` command line options, refer to the `man page <https://www.open-mpi.org/doc/current/man1/mpirun.1.php>`_.
+        working_dir: An absolute path to a directory in which intermediate files can be written to and read from. If
+            no working directory is provided, the system temporary files directory will be used.
+        spin_sq: Target value for the total spin squared for the ground state. If ``None``, no spin will be imposed.
+        clean_working_dir: A flag indicating whether to remove the intermediate files used by the ``Dice``
+            command line application. If ``False``, the intermediate files will be left in a temporary directory in the
+            ``working_dir``.
+
+    Returns:
+        Minimum energy from SCI calculation, SCI coefficients, and average orbital occupancy for spin-up and spin-down orbitals
+    """
+    # Convert bitstring matrix to integer determinants for spin-up/down
+    ci_strs = bitstring_matrix_to_ci_strs(bitstring_matrix)
+    num_configurations = len(ci_strs[0])
+    num_up = bin(ci_strs[0][0])[2:].count("1")
+    num_dn = bin(ci_strs[1][0])[2:].count("1")
+
+    # Set up the working directory
+    working_dir = working_dir or tempfile.gettempdir()
+    intermediate_dir = Path(tempfile.mkdtemp(prefix="dice_cli_files_", dir=working_dir))
+
+    # Write the integrals out as an FCI dump for Dice command line app
+    active_space_path = os.path.join(intermediate_dir, "fcidump.txt")
+    num_orbitals = hcore.shape[0]
+    tools.fcidump.from_integrals(
+        active_space_path, hcore, eri, num_orbitals, (num_up + num_dn)
+    )
+
+    _write_input_files(
+        ci_strs,
+        active_space_path,
+        num_up,
+        num_dn,
+        num_configurations,
+        intermediate_dir,
+        spin_sq,
+        1,
```

Comment:
@kevinsung , is the the takeaway from the max_iter thing? We just hard code this to 1 for SQD workflows since we just want to run one round of Davidson diagonalization + projection

Reply:
I believe this is correct, but I wouldn't be totally confident without a unit test.